bundles/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/kostalinverter/internal/secondgeneration/SecondGenerationHandler.java

```diff
@@ -0,0 +1,410 @@
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.kostalinverter.internal.secondgeneration;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import javax.measure.Unit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ * The {@link SecondGenerationHandler} is responsible for handling commands, which are
+ * sent to one of the channels, and initiation and refreshing regarded to second generation part of the binding.
+ *
+ *
+ * @author Christian Schneider - Initial contribution
+ * @author Christoph Weitkamp - Incorporated new QuantityType (Units of Measurement)
+ * @author Örjan Backsell - Redesigned regarding Piko1020, Piko New Generation
+ */
+@NonNullByDefault
+public class SecondGenerationHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SecondGenerationHandler.class);
+
+    private @Nullable ScheduledFuture<?> secondGenerationPoller;
+
+    private final HttpClient httpClient;
+
+    private List<SecondGenerationChannelConfiguration> channelConfigs = new ArrayList<>();
+    private List<SecondGenerationChannelConfiguration> channelConfigsExt = new ArrayList<>();
+    private List<SecondGenerationChannelConfiguration> channelConfigsExtExt = new ArrayList<>();
+    private List<SecondGenerationChannelConfiguration> channelConfigsConfigurable = new ArrayList<>();
+    private List<SecondGenerationChannelConfiguration> channelConfigsAll = new ArrayList<>();
+
+    private @Nullable SecondGenerationInverterConfig inverterConfig;
+
+    private @Nullable Gson gson;
+
+    public SecondGenerationHandler(Thing thing, HttpClient httpClient) {
+        super(thing);
+        this.httpClient = httpClient;
+    }
+
+    @SuppressWarnings("null")
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String url = inverterConfig.url;
```

Comment:
The field `inverterConfig` can be null at this point. To mitigate this, you can store the field to a local variable before checking it:
```

Future<?> localInverterConfig = inverterConfig;
if (localInverterConfig != null) {
    localInverterConfig .cancel(true);
}
```

Then, you can remove the supression.